tools/runner

```diff
@@ -119,11 +120,14 @@ try:
     logger.info("Running {}/{}"
                 .format(args.runner, args.test))
 
+    started = time.time()
     output, rc = runner_obj.run(tmp_dir, test_params)
```

Comment:
> Is there a way to extract the running time from the process ? Essentially what we'd get if we start the external process with /usr/bin/time (...)
We cannot do that in any easy and consistent way. Currently we allow the `run` callback to be overloaded and that's what we do, see: https://github.com/SymbiFlow/sv-tests/blob/fe3f6eb65523f63aece246fdacf62d76ff8a5c59/tools/runners/tree_sitter_verilog.py#L32

In this case we don't even call any system tool so there is no way we can call it through a call to `/usr/bin/time`

Reply:
> While at it: we could consider recording additional metrics: maximum memory used ? Some parsers are memory hogs, so this might help getting this on the screen.

Is there any particular way what would you like to collect and how should this be displayed/summarized?

Reply:
Initially, it would probably be sufficient to just output the summary data (runtime for everything) at the bottom of the table (and in each individual test-log as you already do in the log-template).

As for the individual data to be evaluated, it would probably good to have that initially in the CSV file for separate processing.
In the current CSV schema, we have rows per test and columns per tool, but to add more data such as time and possibly memory at some point, the columns would start to get less useful (would we have three columns per tool ? With each new metric it would grow in width and is not compatible anymore to previous CSV outputs).
So we probably want to normalize this with rows that contain a column each with test and tool and then the metrics we recorded for it, success, runtime, (and memory if available ... and possibly further metrics we come up with later) in the following columns.

So something roughly like this
```
testname,          testfile,      tags,        tool,  success, time
number-test-42, /path/to/test.sv, 1.2.3 4.5.6, Yosys, true, 0.853
number-test-42, /path/to/test.sv, 1.2.3 4.5.6, Verilator, true, 0.555

```
This makes the CSV file makes it consistent, easy to extend but would require changing aggregation scripts, but since we have not set in stone yet the CSV format, that is fine.